errors when building the schematics during 'npm install'

<!-- To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.
-->

**Actual Behavior**

When running `npm install` on PWA versions before 0.24.0 the following errors are logged in the `postinstall` step `npm --silent run build:schematics`. This interrupts the `npm install` process so no working PWA can be compiled.

```
intershop-pwa-0.23.0>npm install

> intershop-pwa@0.23.0 postinstall D:\DEV_PWA\intershop-pwa-0.23.0
> postinstall && npm --silent run build:tslint-rules && npm --silent run build:schematics && npm run synchronize-lazy-components && node scripts/init-local-environment

up to date in 0.692s
found 0 vulnerabilities

+ json-schema-to-typescript@10.0.0
updated 1 package in 8.975s
found 0 vulnerabilities

creating schemas
Done!
src/extension/factory.ts:26:5 - error TS2322: Type '{ extension: string; path: string; name?: string; restricted?: boolean; project?: string; }' is not assignable to type 'PWAExtensionOptionsSchema'.
  Types of property 'restricted' are incompatible.
    Type 'boolean' is not assignable to type 'never'.

26     options = await detectExtension('extension', host, options);
       ~~~~~~~

src/extension/factory.ts:27:5 - error TS2322: Type '{ name: string; path: string; project?: string; restricted?: boolean; flat?: boolean; artifactFolder?: boolean; }' is not assignable to type 'PWAExtensionOptionsSchema'.
  Types of property 'restricted' are incompatible.
    Type 'boolean' is not assignable to type 'never'.

27     options = await applyNameAndPath('extension', host, options);
       ~~~~~~~

src/extension/factory.ts:28:5 - error TS2322: Type '{ moduleImportPath: any; artifactName: any; project?: string; name?: string; path?: string; restricted?: boolean; flat?: boolean; artifactFolder?: boolean; }' is not assignable to type 'PWAExtensionOptionsSchema'.
  Types of property 'restricted' are incompatible.
    Type 'boolean' is not assignable to type 'never'.

28     options = determineArtifactName('extension', host, options);
       ~~~~~~~

src/model/factory.ts:25:5 - error TS2322: Type '{ extension: string; path: string; name?: string; restricted?: boolean; project?: string; }' is not assignable to type 'PWAModelOptionsSchema'.
  Types of property 'restricted' are incompatible.
    Type 'boolean' is not assignable to type 'never'.

25     options = await detectExtension('model', host, options);
       ~~~~~~~

src/model/factory.ts:26:5 - error TS2322: Type '{ name: string; path: string; project?: string; restricted?: boolean; flat?: boolean; artifactFolder?: boolean; }' is not assignable to type 'PWAModelOptionsSchema'.
  Types of property 'restricted' are incompatible.
    Type 'boolean' is not assignable to type 'never'.

26     options = await applyNameAndPath('model', host, options);
       ~~~~~~~

src/model/factory.ts:27:5 - error TS2322: Type '{ moduleImportPath: any; artifactName: any; project?: string; name?: string; path?: string; restricted?: boolean; flat?: boolean; artifactFolder?: boolean; }' is not assignable to type 'PWAModelOptionsSchema'.
  Types of property 'restricted' are incompatible.
    Type 'boolean' is not assignable to type 'never'.

27     options = determineArtifactName('model', host, options);
       ~~~~~~~

src/page/factory.ts:128:5 - error TS2322: Type '{ routingModule: Path; child: string; name?: string; project?: string; extension?: string; lazy?: boolean; }' is not assignable to type 'PWAPageOptionsSchema'.
  Types of property 'lazy' are incompatible.
    Type 'boolean' is not assignable to type 'never'.

128     options = await determineRoutingModule(host, options);
        ~~~~~~~

src/utils/ts-morph.ts:12:81 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'BufferEncoding'.

12       readFileSync: (p, encoding) => (host.read(p) || readFileSync(p)).toString(encoding),
                                                                                   ~~~~~~~~
Found 8 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! intershop-pwa@0.23.0 postinstall: `postinstall && npm --silent run build:tslint-rules && npm --silent run build:schematics && npm run synchronize-lazy-components && node scripts/init-local-environment`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the intershop-pwa@0.23.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Cause**
A new incompatible version 10.0.0 of `json-schema-to-typescript` was released that will be used since there is currently no configuration that enforces a specific version of this package (https://www.npmjs.com/package/json-schema-to-typescript?activeTab=versions).

**Quick Fix**
~~Replace the following line in `schematics/package.json` (https://github.com/intershop/intershop-pwa/blob/0.23.0/schematics/package.json#L11)~~
```
"prebuild": "npm i --no-save json-schema-to-typescript && npm run generate:schemas",
```
~~with this line~~
```
"prebuild": "npm i --no-save json-schema-to-typescript@9.1.1 && npm run generate:schemas",
```

For fixing please cherry-pick the commit from #475

--- a/schematics/src/extension/schema.json
+++ b/schematics/src/extension/schema.json
@@ -18,7 +18,7 @@
     },
     "restricted": {
       "type": "boolean",
-      "default": "true",
+      "default": true,
       "visible": "false"
     },
     "name": {

--- a/schematics/src/model/schema.json
+++ b/schematics/src/model/schema.json
@@ -17,7 +17,7 @@
     },
     "restricted": {
       "type": "boolean",
-      "default": "true",
+      "default": true,
       "visible": "false"
     },
     "name": {

--- a/schematics/src/page/schema.json
+++ b/schematics/src/page/schema.json
@@ -42,7 +42,7 @@
     "lazy": {
       "type": "boolean",
       "description": "Lazy-load the created page. (default is true)",
-      "default": "true"
+      "default": true
     }
   }
 }

--- a/schematics/src/utils/ts-morph.ts
+++ b/schematics/src/utils/ts-morph.ts
@@ -9,7 +9,7 @@ export function createTsMorphProject(host: Tree) {
       getCurrentDirectory: () => '',
       directoryExistsSync: p => host.exists(p) || existsSync(p),
       fileExistsSync: p => host.exists(p) || existsSync(p),
-      readFileSync: (p, encoding) => (host.read(p) || readFileSync(p)).toString(encoding),
+      readFileSync: (p, encoding) => (host.read(p) || readFileSync(p)).toString(encoding as BufferEncoding),
     } as FileSystemHost,
   });
 }

